code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/AEPPushNotificationBuilder.kt

```diff
@@ -0,0 +1,977 @@
+        setElementColor(
+            smallLayout,
+            R.id.notification_body,
+            "#$expandedBodyTextColor",
+            PushTemplateConstants.MethodNames.SET_TEXT_COLOR,
+            PushTemplateConstants.FriendlyViewNames.NOTIFICATION_BODY_TEXT
+        )
+        setElementColor(
+            expandedLayout,
+            R.id.notification_body_expanded,
+            "#$expandedBodyTextColor",
+            PushTemplateConstants.MethodNames.SET_TEXT_COLOR,
+            PushTemplateConstants.FriendlyViewNames.NOTIFICATION_BODY_TEXT
+        )
+    }
+
+    /**
+     * Sets a provided color hex string to a UI element contained in a specified [RemoteViews]
+     * view.
+     *
+     * @param remoteView `RemoteViews` object containing a UI element to be updated
+     * @param elementId [Int] containing the resource id of the UI element
+     * @param colorHex [String] containing the color hex string
+     * @param methodName `String` containing the method to be called on the UI element to
+     * update the color
+     * @param viewFriendlyName `String` containing the friendly name of the view to be used
+     * for logging purposes
+     */
+    private fun setElementColor(
+        remoteView: RemoteViews,
+        elementId: Int,
+        colorHex: String,
+        methodName: String,
+        viewFriendlyName: String
+    ) {
+        if (colorHex.isEmpty()) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Empty color hex string found, custom color will not be applied to $viewFriendlyName."
+            )
+            return
+        }
+
+        try {
+            remoteView.setInt(elementId, methodName, Color.parseColor(colorHex))
+        } catch (exception: IllegalArgumentException) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Unrecognized hex string passed to Color.parseColor(), custom color will not be applied to $viewFriendlyName."
+            )
+        }
+    }
+
+    /**
+     * Creates a click intent for the specified [Intent] action. This intent is used to handle interactions
+     * with the skip left and skip right buttons in a filmstrip or manual carousel push template notification.
+     *
+     * @param context the application [Context]
+     * @param pushTemplate the [ManualCarouselPushTemplate] object containing the manual carousel push template data
+     * @param intentAction [String] containing the intent action
+     * @param broadcastReceiverClass the [Class] of the broadcast receiver to set in the created pending intent
+     * @param downloadedImageUris [List] of String` containing the downloaded image URIs
+     * @param imageCaptions `List` of String` containing the image captions
+     * @param imageClickActions `List` of String` containing the image click actions
+     * @return the created click [Intent]
+     */
+    internal fun createClickIntent(
```

Comment:
Any reason this is defined in `AEPPushNotificationBuilder` and not `ManualCarouselNotificationBuilder`?

Reply:
nit: I prefer reorganizing the non-private methods, except construct. They can either be moved to a utility class or defined as extensions for appropriate classes.

Reply:
I prefer the extension function approach too, to avoid passing around additional params. I will leave this one open as well to discuss again with @rymorale and @prudrabhat